include/armadillo_bits/diskio_meat.hpp

```diff
@@ -1319,7 +1331,10 @@ diskio::save_hdf5_binary(const Mat<eT>& x, const hdf5_name& spec)
     for (size_t i = 0; i < groups.size(); ++i)  { arma_H5Gclose(groups[i]); }
     arma_H5Fclose(file);
     
-    if(save_okay == true) { save_okay = diskio::safe_rename(tmp_name, spec.filename); }
+    if (save_okay == true)
+      {
+      if ((file_is_hdf5 != 1) || (spec.append == false)) { save_okay = diskio::safe_rename(tmp_name, spec.filename); }
+      }
     
     return save_okay;
     }
```

Comment:
The `diskio::save_hdf5_binary(const Mat<eT>& x, const hdf5_name& spec)` function was extended to support appending.  The documentation for `.save()` applies to both matrices and cubes, so the corresponding function to handle cubes also needs to be extended: `diskio::save_hdf5_binary(const Cube<eT>& x, const hdf5_name& spec)`